### ISSUE TITLE
pedestrian detection

### ISSUE THREAD

**Issue by ghostjat** [geekgirljoy/OCR_Neural_Network#1]
Is it possible pedestrian detection using HOG features extraction in fann.

**Comment by geekgirljoy** [comment]
Hello ghostjat,

Yes, you should be able to use collections of histogram gradients like HoG features.

**However, several things:**

1. FANN just provides the neural network structure and training algorithms / backwards propagation etc... It does not have "helper" software-features that will extract the image-features to training/input data for you. 

As such, you will have to pre-process your images to generate the histograms and save them in the FANN .data training file format.

It appears that you are a PHP developer so your immediate options for extracting and processing your pixel data are:

**GD-LIB:** ( https://www.php.net/manual/en/book.image.php ) which is what I use and demonstrate. It's simple and low level, which means it tends to be the faster option but it has few "extras". It is usually installed along with PHP in most cases.

**ImageMagick:** Offers a more "advanced" / "robust" set of tools. However it can be slower and isn't usually installed with PHP by default.

**If you decide to use ImageMagick over the GD-LIB here are the install instructions:** (https://www.php.net/manual/en/imagick.setup.php )



2. If your purpose is "Object Detection/Identification" then you might want to look at YOLO over HoG which has been shown to be faster and tends to be more accurate.

**YOLO paper on arXiv:** ( https://arxiv.org/abs/1506.02640 )



3. I am certain you know what you are doing but I always like to offer a word of caution when there is the possibility of danger to life/limb as might be the case if you intend to build a system that for example, detected if the light should be green (safe for cars to drive) if there are no people in the crosswalk.

I'd ask you to step back for a moment and ask yourself if you actually need to identify if:

**A:** there is "a person" blocking the cross walk (light should not be green)

or

**B:** There is an obstacle in the crosswalk (light should not be green)

This might seem like the same thing but it's not.

In the case of A, it is far more prone to failure that can be dangerous to the driver or people in the crosswalk.

A system with the limited goal of detecting if a "person" is present would not operate properly if, for example, a box fell off a truck or a baby stroller rolled out into the view of the system.

Since it was trained to look for "people" shapes System A would eventually lead to loss of life and damage of property.

System B however is essentially trained to notice anything that deviates from it's "ideal norm" (an empty cross walk) without regard to the shape of the obstacle and as such, should behave as expected in the case of unexpected obstacles in the area of interest.

Object Identification should be of a lessor priority than Object Detection if there is the possibility of danger to human life as the result of your system.


Hope this helps!

**Comment by geekgirljoy** [comment]
Closing this question/issue.
